digraph "unitGraph" {
    "Block 0:
[preds: ] [succs: 5 37 1 ]
r13 := @parameter0: java.lang.String[];
i0 = 0;
"
    "Block 1:
[preds: 0 ] [succs: 5 37 2 ]
$r0 = new java.util.PropertyPermission;
"
    "Block 2:
[preds: 1 ] [succs: 5 37 3 ]
specialinvoke $r0.<java.util.PropertyPermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"
    "Block 3:
[preds: 2 ] [succs: 5 37 4 ]
r14 = $r0;
"
    "Block 4:
[preds: 3 ] [succs: 12 37 8 ]
goto [?= $r1 = new java.io.FilePermission];
"
    "Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
"
    "Block 6:
[preds: 5 ] [succs: 7 37 ]
r15 = $r12;
"
    "Block 7:
[preds: 6 ] [succs: 12 37 8 ]
i0 = i0 + 1;
"
    "Block 8:
[preds: 4 7 ] [succs: 12 37 9 ]
$r1 = new java.io.FilePermission;
"
    "Block 9:
[preds: 8 ] [succs: 12 37 10 ]
specialinvoke $r1.<java.io.FilePermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"
    "Block 10:
[preds: 9 ] [succs: 12 37 11 ]
r16 = $r1;
"
    "Block 11:
[preds: 10 ] [succs: 19 37 15 ]
goto [?= $r2 = new java.net.SocketPermission];
"
    "Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
"
    "Block 13:
[preds: 12 ] [succs: 14 37 ]
r17 = $r11;
"
    "Block 14:
[preds: 13 ] [succs: 19 37 15 ]
i0 = i0 + 1;
"
    "Block 15:
[preds: 11 14 ] [succs: 19 37 16 ]
$r2 = new java.net.SocketPermission;
"
    "Block 16:
[preds: 15 ] [succs: 19 37 17 ]
specialinvoke $r2.<java.net.SocketPermission: void <init>(java.lang.String,java.lang.String)>(null, \"connect\");
"
    "Block 17:
[preds: 16 ] [succs: 19 37 18 ]
r18 = $r2;
"
    "Block 18:
[preds: 17 ] [succs: 26 37 22 ]
goto [?= $r3 = new java.lang.RuntimePermission];
"
    "Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
"
    "Block 20:
[preds: 19 ] [succs: 21 37 ]
r19 = $r10;
"
    "Block 21:
[preds: 20 ] [succs: 26 37 22 ]
i0 = i0 + 1;
"
    "Block 22:
[preds: 18 21 ] [succs: 26 37 23 ]
$r3 = new java.lang.RuntimePermission;
"
    "Block 23:
[preds: 22 ] [succs: 26 37 24 ]
specialinvoke $r3.<java.lang.RuntimePermission: void <init>(java.lang.String)>(null);
"
    "Block 24:
[preds: 23 ] [succs: 26 37 25 ]
r20 = $r3;
"
    "Block 25:
[preds: 24 ] [succs: 33 37 29 ]
goto [?= $r4 = new java.security.UnresolvedPermission];
"
    "Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
"
    "Block 27:
[preds: 26 ] [succs: 28 37 ]
r21 = $r9;
"
    "Block 28:
[preds: 27 ] [succs: 33 37 29 ]
i0 = i0 + 1;
"
    "Block 29:
[preds: 25 28 ] [succs: 33 37 30 ]
$r4 = new java.security.UnresolvedPermission;
"
    "Block 30:
[preds: 29 ] [succs: 33 37 31 ]
specialinvoke $r4.<java.security.UnresolvedPermission: void <init>(java.lang.String,java.lang.String,java.lang.String,java.security.cert.Certificate[])>(null, \"blah\", \"read\", null);
"
    "Block 31:
[preds: 30 ] [succs: 33 37 32 ]
r22 = $r4;
"
    "Block 32:
[preds: 31 ] [succs: 36 37 ]
goto [?= (branch)];
"
    "Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
"
    "Block 34:
[preds: 33 ] [succs: 35 37 ]
r23 = $r8;
"
    "Block 35:
[preds: 34 ] [succs: 36 37 ]
i0 = i0 + 1;
"
    "Block 36:
[preds: 32 35 ] [succs: 38 ]
goto [?= (branch)];
"
    "Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
"
    "Block 38:
[preds: 36 ] [succs: 39 40 ]
if i0 == 5 goto return;
"
    "Block 39:
[preds: 38 ] [succs: ]
$r5 = new java.lang.Exception;
specialinvoke $r5.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: didn\'t catch enough NullPointerExceptions\");
throw $r5;
"
    "Block 40:
[preds: 38 ] [succs: ]
return;
"
    "Block 0:
[preds: ] [succs: 5 37 1 ]
r13 := @parameter0: java.lang.String[];
i0 = 0;
"->"Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
";
    "Block 0:
[preds: ] [succs: 5 37 1 ]
r13 := @parameter0: java.lang.String[];
i0 = 0;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 0:
[preds: ] [succs: 5 37 1 ]
r13 := @parameter0: java.lang.String[];
i0 = 0;
"->"Block 1:
[preds: 0 ] [succs: 5 37 2 ]
$r0 = new java.util.PropertyPermission;
";
    "Block 1:
[preds: 0 ] [succs: 5 37 2 ]
$r0 = new java.util.PropertyPermission;
"->"Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
";
    "Block 1:
[preds: 0 ] [succs: 5 37 2 ]
$r0 = new java.util.PropertyPermission;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 1:
[preds: 0 ] [succs: 5 37 2 ]
$r0 = new java.util.PropertyPermission;
"->"Block 2:
[preds: 1 ] [succs: 5 37 3 ]
specialinvoke $r0.<java.util.PropertyPermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
";
    "Block 2:
[preds: 1 ] [succs: 5 37 3 ]
specialinvoke $r0.<java.util.PropertyPermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"->"Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
";
    "Block 2:
[preds: 1 ] [succs: 5 37 3 ]
specialinvoke $r0.<java.util.PropertyPermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 2:
[preds: 1 ] [succs: 5 37 3 ]
specialinvoke $r0.<java.util.PropertyPermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"->"Block 3:
[preds: 2 ] [succs: 5 37 4 ]
r14 = $r0;
";
    "Block 3:
[preds: 2 ] [succs: 5 37 4 ]
r14 = $r0;
"->"Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
";
    "Block 3:
[preds: 2 ] [succs: 5 37 4 ]
r14 = $r0;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 3:
[preds: 2 ] [succs: 5 37 4 ]
r14 = $r0;
"->"Block 4:
[preds: 3 ] [succs: 12 37 8 ]
goto [?= $r1 = new java.io.FilePermission];
";
    "Block 4:
[preds: 3 ] [succs: 12 37 8 ]
goto [?= $r1 = new java.io.FilePermission];
"->"Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
";
    "Block 4:
[preds: 3 ] [succs: 12 37 8 ]
goto [?= $r1 = new java.io.FilePermission];
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 4:
[preds: 3 ] [succs: 12 37 8 ]
goto [?= $r1 = new java.io.FilePermission];
"->"Block 8:
[preds: 4 7 ] [succs: 12 37 9 ]
$r1 = new java.io.FilePermission;
";
    "Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
"->"Block 6:
[preds: 5 ] [succs: 7 37 ]
r15 = $r12;
";
    "Block 5:
[preds: 0 1 2 3 ] [succs: 6 37 ]
$r12 := @caughtexception;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 6:
[preds: 5 ] [succs: 7 37 ]
r15 = $r12;
"->"Block 7:
[preds: 6 ] [succs: 12 37 8 ]
i0 = i0 + 1;
";
    "Block 6:
[preds: 5 ] [succs: 7 37 ]
r15 = $r12;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 7:
[preds: 6 ] [succs: 12 37 8 ]
i0 = i0 + 1;
"->"Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
";
    "Block 7:
[preds: 6 ] [succs: 12 37 8 ]
i0 = i0 + 1;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 7:
[preds: 6 ] [succs: 12 37 8 ]
i0 = i0 + 1;
"->"Block 8:
[preds: 4 7 ] [succs: 12 37 9 ]
$r1 = new java.io.FilePermission;
";
    "Block 8:
[preds: 4 7 ] [succs: 12 37 9 ]
$r1 = new java.io.FilePermission;
"->"Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
";
    "Block 8:
[preds: 4 7 ] [succs: 12 37 9 ]
$r1 = new java.io.FilePermission;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 8:
[preds: 4 7 ] [succs: 12 37 9 ]
$r1 = new java.io.FilePermission;
"->"Block 9:
[preds: 8 ] [succs: 12 37 10 ]
specialinvoke $r1.<java.io.FilePermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
";
    "Block 9:
[preds: 8 ] [succs: 12 37 10 ]
specialinvoke $r1.<java.io.FilePermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"->"Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
";
    "Block 9:
[preds: 8 ] [succs: 12 37 10 ]
specialinvoke $r1.<java.io.FilePermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 9:
[preds: 8 ] [succs: 12 37 10 ]
specialinvoke $r1.<java.io.FilePermission: void <init>(java.lang.String,java.lang.String)>(null, \"read\");
"->"Block 10:
[preds: 9 ] [succs: 12 37 11 ]
r16 = $r1;
";
    "Block 10:
[preds: 9 ] [succs: 12 37 11 ]
r16 = $r1;
"->"Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
";
    "Block 10:
[preds: 9 ] [succs: 12 37 11 ]
r16 = $r1;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 10:
[preds: 9 ] [succs: 12 37 11 ]
r16 = $r1;
"->"Block 11:
[preds: 10 ] [succs: 19 37 15 ]
goto [?= $r2 = new java.net.SocketPermission];
";
    "Block 11:
[preds: 10 ] [succs: 19 37 15 ]
goto [?= $r2 = new java.net.SocketPermission];
"->"Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
";
    "Block 11:
[preds: 10 ] [succs: 19 37 15 ]
goto [?= $r2 = new java.net.SocketPermission];
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 11:
[preds: 10 ] [succs: 19 37 15 ]
goto [?= $r2 = new java.net.SocketPermission];
"->"Block 15:
[preds: 11 14 ] [succs: 19 37 16 ]
$r2 = new java.net.SocketPermission;
";
    "Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
"->"Block 13:
[preds: 12 ] [succs: 14 37 ]
r17 = $r11;
";
    "Block 12:
[preds: 4 7 8 9 10 ] [succs: 13 37 ]
$r11 := @caughtexception;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 13:
[preds: 12 ] [succs: 14 37 ]
r17 = $r11;
"->"Block 14:
[preds: 13 ] [succs: 19 37 15 ]
i0 = i0 + 1;
";
    "Block 13:
[preds: 12 ] [succs: 14 37 ]
r17 = $r11;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 14:
[preds: 13 ] [succs: 19 37 15 ]
i0 = i0 + 1;
"->"Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
";
    "Block 14:
[preds: 13 ] [succs: 19 37 15 ]
i0 = i0 + 1;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 14:
[preds: 13 ] [succs: 19 37 15 ]
i0 = i0 + 1;
"->"Block 15:
[preds: 11 14 ] [succs: 19 37 16 ]
$r2 = new java.net.SocketPermission;
";
    "Block 15:
[preds: 11 14 ] [succs: 19 37 16 ]
$r2 = new java.net.SocketPermission;
"->"Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
";
    "Block 15:
[preds: 11 14 ] [succs: 19 37 16 ]
$r2 = new java.net.SocketPermission;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 15:
[preds: 11 14 ] [succs: 19 37 16 ]
$r2 = new java.net.SocketPermission;
"->"Block 16:
[preds: 15 ] [succs: 19 37 17 ]
specialinvoke $r2.<java.net.SocketPermission: void <init>(java.lang.String,java.lang.String)>(null, \"connect\");
";
    "Block 16:
[preds: 15 ] [succs: 19 37 17 ]
specialinvoke $r2.<java.net.SocketPermission: void <init>(java.lang.String,java.lang.String)>(null, \"connect\");
"->"Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
";
    "Block 16:
[preds: 15 ] [succs: 19 37 17 ]
specialinvoke $r2.<java.net.SocketPermission: void <init>(java.lang.String,java.lang.String)>(null, \"connect\");
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 16:
[preds: 15 ] [succs: 19 37 17 ]
specialinvoke $r2.<java.net.SocketPermission: void <init>(java.lang.String,java.lang.String)>(null, \"connect\");
"->"Block 17:
[preds: 16 ] [succs: 19 37 18 ]
r18 = $r2;
";
    "Block 17:
[preds: 16 ] [succs: 19 37 18 ]
r18 = $r2;
"->"Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
";
    "Block 17:
[preds: 16 ] [succs: 19 37 18 ]
r18 = $r2;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 17:
[preds: 16 ] [succs: 19 37 18 ]
r18 = $r2;
"->"Block 18:
[preds: 17 ] [succs: 26 37 22 ]
goto [?= $r3 = new java.lang.RuntimePermission];
";
    "Block 18:
[preds: 17 ] [succs: 26 37 22 ]
goto [?= $r3 = new java.lang.RuntimePermission];
"->"Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
";
    "Block 18:
[preds: 17 ] [succs: 26 37 22 ]
goto [?= $r3 = new java.lang.RuntimePermission];
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 18:
[preds: 17 ] [succs: 26 37 22 ]
goto [?= $r3 = new java.lang.RuntimePermission];
"->"Block 22:
[preds: 18 21 ] [succs: 26 37 23 ]
$r3 = new java.lang.RuntimePermission;
";
    "Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
"->"Block 20:
[preds: 19 ] [succs: 21 37 ]
r19 = $r10;
";
    "Block 19:
[preds: 11 14 15 16 17 ] [succs: 20 37 ]
$r10 := @caughtexception;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 20:
[preds: 19 ] [succs: 21 37 ]
r19 = $r10;
"->"Block 21:
[preds: 20 ] [succs: 26 37 22 ]
i0 = i0 + 1;
";
    "Block 20:
[preds: 19 ] [succs: 21 37 ]
r19 = $r10;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 21:
[preds: 20 ] [succs: 26 37 22 ]
i0 = i0 + 1;
"->"Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
";
    "Block 21:
[preds: 20 ] [succs: 26 37 22 ]
i0 = i0 + 1;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 21:
[preds: 20 ] [succs: 26 37 22 ]
i0 = i0 + 1;
"->"Block 22:
[preds: 18 21 ] [succs: 26 37 23 ]
$r3 = new java.lang.RuntimePermission;
";
    "Block 22:
[preds: 18 21 ] [succs: 26 37 23 ]
$r3 = new java.lang.RuntimePermission;
"->"Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
";
    "Block 22:
[preds: 18 21 ] [succs: 26 37 23 ]
$r3 = new java.lang.RuntimePermission;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 22:
[preds: 18 21 ] [succs: 26 37 23 ]
$r3 = new java.lang.RuntimePermission;
"->"Block 23:
[preds: 22 ] [succs: 26 37 24 ]
specialinvoke $r3.<java.lang.RuntimePermission: void <init>(java.lang.String)>(null);
";
    "Block 23:
[preds: 22 ] [succs: 26 37 24 ]
specialinvoke $r3.<java.lang.RuntimePermission: void <init>(java.lang.String)>(null);
"->"Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
";
    "Block 23:
[preds: 22 ] [succs: 26 37 24 ]
specialinvoke $r3.<java.lang.RuntimePermission: void <init>(java.lang.String)>(null);
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 23:
[preds: 22 ] [succs: 26 37 24 ]
specialinvoke $r3.<java.lang.RuntimePermission: void <init>(java.lang.String)>(null);
"->"Block 24:
[preds: 23 ] [succs: 26 37 25 ]
r20 = $r3;
";
    "Block 24:
[preds: 23 ] [succs: 26 37 25 ]
r20 = $r3;
"->"Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
";
    "Block 24:
[preds: 23 ] [succs: 26 37 25 ]
r20 = $r3;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 24:
[preds: 23 ] [succs: 26 37 25 ]
r20 = $r3;
"->"Block 25:
[preds: 24 ] [succs: 33 37 29 ]
goto [?= $r4 = new java.security.UnresolvedPermission];
";
    "Block 25:
[preds: 24 ] [succs: 33 37 29 ]
goto [?= $r4 = new java.security.UnresolvedPermission];
"->"Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
";
    "Block 25:
[preds: 24 ] [succs: 33 37 29 ]
goto [?= $r4 = new java.security.UnresolvedPermission];
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 25:
[preds: 24 ] [succs: 33 37 29 ]
goto [?= $r4 = new java.security.UnresolvedPermission];
"->"Block 29:
[preds: 25 28 ] [succs: 33 37 30 ]
$r4 = new java.security.UnresolvedPermission;
";
    "Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
"->"Block 27:
[preds: 26 ] [succs: 28 37 ]
r21 = $r9;
";
    "Block 26:
[preds: 18 21 22 23 24 ] [succs: 27 37 ]
$r9 := @caughtexception;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 27:
[preds: 26 ] [succs: 28 37 ]
r21 = $r9;
"->"Block 28:
[preds: 27 ] [succs: 33 37 29 ]
i0 = i0 + 1;
";
    "Block 27:
[preds: 26 ] [succs: 28 37 ]
r21 = $r9;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 28:
[preds: 27 ] [succs: 33 37 29 ]
i0 = i0 + 1;
"->"Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
";
    "Block 28:
[preds: 27 ] [succs: 33 37 29 ]
i0 = i0 + 1;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 28:
[preds: 27 ] [succs: 33 37 29 ]
i0 = i0 + 1;
"->"Block 29:
[preds: 25 28 ] [succs: 33 37 30 ]
$r4 = new java.security.UnresolvedPermission;
";
    "Block 29:
[preds: 25 28 ] [succs: 33 37 30 ]
$r4 = new java.security.UnresolvedPermission;
"->"Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
";
    "Block 29:
[preds: 25 28 ] [succs: 33 37 30 ]
$r4 = new java.security.UnresolvedPermission;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 29:
[preds: 25 28 ] [succs: 33 37 30 ]
$r4 = new java.security.UnresolvedPermission;
"->"Block 30:
[preds: 29 ] [succs: 33 37 31 ]
specialinvoke $r4.<java.security.UnresolvedPermission: void <init>(java.lang.String,java.lang.String,java.lang.String,java.security.cert.Certificate[])>(null, \"blah\", \"read\", null);
";
    "Block 30:
[preds: 29 ] [succs: 33 37 31 ]
specialinvoke $r4.<java.security.UnresolvedPermission: void <init>(java.lang.String,java.lang.String,java.lang.String,java.security.cert.Certificate[])>(null, \"blah\", \"read\", null);
"->"Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
";
    "Block 30:
[preds: 29 ] [succs: 33 37 31 ]
specialinvoke $r4.<java.security.UnresolvedPermission: void <init>(java.lang.String,java.lang.String,java.lang.String,java.security.cert.Certificate[])>(null, \"blah\", \"read\", null);
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 30:
[preds: 29 ] [succs: 33 37 31 ]
specialinvoke $r4.<java.security.UnresolvedPermission: void <init>(java.lang.String,java.lang.String,java.lang.String,java.security.cert.Certificate[])>(null, \"blah\", \"read\", null);
"->"Block 31:
[preds: 30 ] [succs: 33 37 32 ]
r22 = $r4;
";
    "Block 31:
[preds: 30 ] [succs: 33 37 32 ]
r22 = $r4;
"->"Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
";
    "Block 31:
[preds: 30 ] [succs: 33 37 32 ]
r22 = $r4;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 31:
[preds: 30 ] [succs: 33 37 32 ]
r22 = $r4;
"->"Block 32:
[preds: 31 ] [succs: 36 37 ]
goto [?= (branch)];
";
    "Block 32:
[preds: 31 ] [succs: 36 37 ]
goto [?= (branch)];
"->"Block 36:
[preds: 32 35 ] [succs: 38 ]
goto [?= (branch)];
";
    "Block 32:
[preds: 31 ] [succs: 36 37 ]
goto [?= (branch)];
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
"->"Block 34:
[preds: 33 ] [succs: 35 37 ]
r23 = $r8;
";
    "Block 33:
[preds: 25 28 29 30 31 ] [succs: 34 37 ]
$r8 := @caughtexception;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 34:
[preds: 33 ] [succs: 35 37 ]
r23 = $r8;
"->"Block 35:
[preds: 34 ] [succs: 36 37 ]
i0 = i0 + 1;
";
    "Block 34:
[preds: 33 ] [succs: 35 37 ]
r23 = $r8;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 35:
[preds: 34 ] [succs: 36 37 ]
i0 = i0 + 1;
"->"Block 36:
[preds: 32 35 ] [succs: 38 ]
goto [?= (branch)];
";
    "Block 35:
[preds: 34 ] [succs: 36 37 ]
i0 = i0 + 1;
"->"Block 37:
[preds: 0 1 2 3 4 7 8 9 10 11 14 15 16 17 18 21 22 23 24 25 28 29 30 31 5 6 12 13 19 20 26 27 32 33 34 35 ] [succs: ]
$r6 := @caughtexception;
r24 = $r6;
$r7 = new java.lang.Exception;
specialinvoke $r7.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: Wrong exception thrown\");
throw $r7;
";
    "Block 36:
[preds: 32 35 ] [succs: 38 ]
goto [?= (branch)];
"->"Block 38:
[preds: 36 ] [succs: 39 40 ]
if i0 == 5 goto return;
";
    "Block 38:
[preds: 36 ] [succs: 39 40 ]
if i0 == 5 goto return;
"->"Block 39:
[preds: 38 ] [succs: ]
$r5 = new java.lang.Exception;
specialinvoke $r5.<java.lang.Exception: void <init>(java.lang.String)>(\"Test failed: didn\'t catch enough NullPointerExceptions\");
throw $r5;
";
    "Block 38:
[preds: 36 ] [succs: 39 40 ]
if i0 == 5 goto return;
"->"Block 40:
[preds: 38 ] [succs: ]
return;
";
}
